tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -5811,6 +5811,109 @@ def _assert_is_fit(self, message_suffix: str = None):
                 error_message = f"{error_message} `.{message_suffix}`."
             raise AssertionError(error_message)
 
+    def confusion_matrix(
+        self, data=None, normalize=None, labels=None, display=True, save_path=None, **kwargs
+    ):
+        """
+        Compute and optionally plot the confusion matrix for classification models.
+
+        Parameters
+        ----------
+        data : str or TabularDataset, optional
+            Data to evaluate. If None, uses the training or validation data stored in the predictor.
+            Can be a file path or TabularDataset object.
+        normalize : {'true', 'pred', 'all'}, optional
+            Normalization mode (see sklearn.metrics.confusion_matrix). If None, no normalization is applied.
+        labels : list, optional
+            Class label ordering. If None, uses the predictor's known class labels.
+        display : bool, default=True
+            If True, displays the confusion matrix plot.
+        save_path : str, optional
+            File path to save the confusion matrix plot. If None, the plot is not saved.
+        **kwargs : dict
+            Additional keyword arguments passed to sklearn.metrics.confusion_matrix.
+
+        Returns
+        -------
+        cm : ndarray of shape (n_classes, n_classes)
+            Confusion matrix where entry (i, j) indicates the number of samples with true
+            label i and predicted label j.
+
+        Raises
+        ------
+        ValueError
+            If the problem type is not 'binary' or 'multiclass'.
+            If no data is available to compute the confusion matrix.
+            If the label column is missing in the provided data.
+
+        Examples
+        --------
+        >>> predictor = TabularPredictor(label='class').fit(train_data)
+        >>> cm = predictor.confusion_matrix()  # Uses training data
+        >>> cm = predictor.confusion_matrix(test_data)  # Uses test data
+        >>> cm = predictor.confusion_matrix(test_data, normalize='true', save_path='cm.png')
+
+        References
+        ----------
+        https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html
+        """
+        from sklearn.metrics import confusion_matrix as sk_confusion_matrix, ConfusionMatrixDisplay
+        import matplotlib.pyplot as plt
+
+        # Validate problem type
+        if self.problem_type not in ["binary", "multiclass"]:
+            raise ValueError(
+                f"Confusion matrix is only applicable to classification problems. "
+                f"Current problem type: '{self.problem_type}'."
+            )
+
+        # Load or prepare data
+        if data is None:
+            X, y, X_val, y_val = self._trainer.load_data()
```

Comment:
This data is already preprocessed, so we need to skip preprocessing in the `predict` call.